CMSgov/easi-app
# [EASI-1518] Fix type of CedarSystems in GraphQL schema

# EASI-1518

## Changes and Description

- Change the return type of the `cedarSystems` query in our GQL schema from `[CedarSystem]` (a possibly-null array of possibly-null systems) to `[CedarSystem!]!` (a non-null array with non-null elements, though the array can still be empty).
- Cleanup some frontend code using this query, removing some now-unnecessary type assertions.

## How to test this change

- Connect to the VPN.
- Enable CEDAR Core connectivity (the easiest way is to edit `pkg/cedar/core/client.go`; inside `NewClient()`'s definition of `fnEmit()`, have `fnEmit()` just return true.
- Run the app locally.
- Log in to the app with only the EASI_P_USER job code to check that system info from CEDAR is displayed on the home page.
- Go to the Systems tab to check that system info is also displayed on that page.

## Notes
- Frontend team: this also affected the type definitions associated with the `GetCedarSystemsAndBookmarks` query in `src/queries/GetCedarSystemsAndBookmarksQuery.ts`, but that query doesn't appear to be used anywhere except test code; can that query be deleted?
